sklearn/metrics/_ranking.py

```diff
@@ -202,10 +211,16 @@ def _binary_uninterpolated_average_precision(
         return -np.sum(np.diff(recall) * np.array(precision)[:-1])
 
     y_type = type_of_target(y_true)
+
+    # Multiclass is a specific case of the multilabel setting here
+    if y_type == "multiclass":
+        y_true = label_binarize(y_true, classes=np.unique(y_true))
+        y_type = "multilabel-indicator"
```

Comment:
I'd remove this and do the warning below if ``y_type in ['multi-label-indicator', 'multiclass']``